[DIR-1201] add route details page

stefan-kracht opened the pull request:
## Description

- Added a detail page for the routes
- Added e2e test to that routes page

![CleanShot 2024-03-27 at 14 20 14@2x](https://github.com/direktiv/direktiv/assets/121789579/3ba7c52f-8bac-451c-b83d-4895c2a589e4)

- minor changes in the route list
  - the methods will be cropped if there are more than two
  - columns are swaped to be consistent with the details page

| Before ![CleanShot 2024-03-27 at 14 16 25@2x](https://github.com/direktiv/direktiv/assets/121789579/fcf5b853-7d32-490f-bc4b-188e3db3ed67) | After ![CleanShot 2024-03-27 at 14 16 18@2x](https://github.com/direktiv/direktiv/assets/121789579/2d70a5a7-fdf5-4473-adf3-b9a68592c144) |
|--------|--------|

- Some components of the routes list, are also use by the routes details page, therefore I moved them one folder up

## Checklist

- [x] Documentation updated if required
- [x] Test coverage is appropriate
      
## Checklist Internal

- [x] Linear issue linked (e.g. [DIR-XXXX] pull request title)
- [x] Has the PR been labeled

julia-zielke commented on ui/e2e/gateway/routes/details/index.spec.ts:
```diff
@@ -0,0 +1,175 @@
+
+  await expect(
+    page.getByText("received 1 log entry"),
+    "It does have any 1 log entry, after the request to the gateway was made"
```
```suggestion
    "It does have 1 log entry, after the request to the gateway was made"
```

julia-zielke commented on ui/e2e/gateway/routes/details/index.spec.ts:
```diff
@@ -0,0 +1,175 @@
+
+  await expect(
+    page.getByTestId("route-details-header").getByText("no methods set"),
+    "it renders the a note that no methods are set"
```
```suggestion
    "it renders the note that no methods are set"
```

stefan-kracht commented:
Thanks for your suggestions @julia-zielke. I fixed the typos and also changed the icon to the `square-chevron-right`*. This is much better now!



*I was also thinking about the `square-gantt-chart` but it would require an update of the icon package which has some small breaking changes.
